plugin/src/main/kotlin/net/octyl/ts2kt/gradle/tasks/DiscoverTs2ktExecutable.kt

```diff
@@ -118,5 +118,38 @@ open class DiscoverTs2ktExecutable : DefaultTask() {
             }
         }
     }
+}
 
+/**
+ * Information on what to name the ts2kt script and what to put inside, based on the build platform
+ */
+private fun ts2ktScriptDescriptor() = with(OperatingSystem.current()) {
+    when {
+        isWindows -> Script.Bat
+        isUnix -> Script.Sh
+        else -> throw IllegalStateException("Unknown edge case: Platform that is neither Windows not Unix")
```

Comment:
I would use this same logic for the OS handling above.

Reply:
While yes, the code is similar, that would be an instance of "wrong abstraction"

Here we are switching on a platform's scripting backend (sh is supported on all unix platforms), while earlier we were switching on a platform's "start process" capability. 

Though similar, technically different.